tests/test_taskgroups.py

```diff
@@ -80,6 +80,59 @@ async def test_spawn_after_error():
     exc.match('This task group is not active; no new tasks can be spawned')
 
 
+async def test_start_no_value():
+    async def taskfunc(*, task_status):
+        task_status.started()
+
+    async with create_task_group() as tg:
+        value = await tg.start(taskfunc)
+        assert value is None
+
+
+async def test_start_with_value():
+    async def taskfunc(*, task_status):
+        task_status.started('foo')
+
+    async with create_task_group() as tg:
+        value = await tg.start(taskfunc)
+        assert value == 'foo'
+
+
+async def test_start_crash_before_started_call():
+    async def taskfunc(*, task_status):
+        raise Exception('foo')
+
+    async with create_task_group() as tg:
+        with pytest.raises(Exception) as exc:
+            await tg.start(taskfunc)
+
+    exc.match('foo')
+
+
+async def test_start_crash_after_started_call():
+    async def taskfunc(*, task_status):
+        task_status.started(2)
+        raise Exception('foo')
+
+    with pytest.raises(Exception) as exc:
+        async with create_task_group() as tg:
+            value = await tg.start(taskfunc)
+
+    exc.match('foo')
+    assert value == 2
+
+
+async def test_start_no_started_call():
+    async def taskfunc(*, task_status):
+        pass
+
+    async with create_task_group() as tg:
+        with pytest.raises(RuntimeError) as exc:
```

Comment:
```suggestion
        with pytest.raises(RuntimeError, match="Child exited")`

Reply:
Trio starts the exception message with a "c" which is why I left out the first character. I'll probably do a mass refactoring later where I start using `match=...` in `pytest.raises()`.